design-documents/background-processes/background-processes-proposal.md

```diff
@@ -0,0 +1,169 @@
+- security_clean_password_reset_request_event_records
+- magento_newrelicreporting_cron
+- mysqlmq_clean_messages
+- outdated_authentication_failures_cleanup
+- expired_tokens_cleanup
+- catalog_product_alert
+- sitemap_generate
+- backend_clean_cache
+- aggregate_sales_report_shipment_data
+- currency_rates_update
+- captcha_delete_old_attempts
+- captcha_delete_expired_images
+- paypal_fetch_settlement_reports
+- messagequeue_clean_outdated_locks
+- consumers_runner
+- consumerConsumerName
+- bulk_cleanup
+- catalogrule_apply_all
+- newsletter_send_all
+- system_backup
+- visitor_clean
+- analytics_subscribe
+- analytics_update
+- analytics_collect_data
+- aggregate_sales_report_coupons_data
+
+This excludes 3rd party bundled extensions such as DotDigital.
+
+## 2. Visibility
+On Magento 1 we had the AOE Scheduler which provided us a considerably enhanced user experience. This
+module isn't available for Magento 2. Allowing webmasters and developers to clearly see what processes
+are running, scheduled and complete gives them greater control and reduces support requests and increases
+confidence in the platform.
+
+- Admin visible timeline of scheduled tasks (Scheduled, in progress, complete, errored)
+- Ability to schedule, delete, remove, pause specific tasks.
+- Cleanup of redundant processes to reduce noise
+- Logging and Alerting
+
+## 3. Education
+- Technical documentation for effectively implementing schedules into Magento modules
+- Technical documentation for when to use cron vs AMQP
+- Setup documentation for multi threaded implementation  
+
+# Examples
+
+## Example 1 - Rogue process
+A process that connects to an API such as a payment provider could end up getting stuck in a loop
+and blocking any other process from progressing currently there is no mechanic in place to
+a: Allow other business critical processes to continue aside from defining multiple crontab processes in groups
+b: Report or stop the rogue process from continuing until manually spotted
+c: No visibility of the processes aside from checking the database
+d: No reporting or alerting of the above condition
+
+# Staged Approach
+The overhaul of the background processing within Magento is an large project with high impact to to the
+current public API's and implementation of the current Magento Cron.
+
+## Stage 1 - Framework
+Create/expand the framework for AMQP and Worker management.
+
+1. Review queue interfaces
+2. Standardise data structure
+3. Error handling, logging and alerting
+4. Review worker interfaces
+5. Review worker management
+
+###Problems to solve:
+1. Worker management
+How to manage workers at from simple to expert levels of complexity. Currently the cron spawns workers
+with (from the looks of it) no visibility over the threads that are spawned. We also need to factor
+in completely multi-node infrastructure.
+The aim should be to provide a highly scalable implementation from a highly scaled environment to a
```

Comment:
We can give reference to https://kubernetes.io/docs/tasks/job/automated-tasks-with-cron-jobs/ since Kubernetes is preferred orchestration solution.